component/loki/write/write_test.go

```diff
@@ -90,19 +91,22 @@ func TestUnmarshallWalAttrributes(t *testing.T) {
 				MaxSegmentAge:    wal.DefaultMaxSegmentAge,
 				MinReadFrequency: wal.DefaultWatchConfig.MinReadFrequency,
 				MaxReadFrequency: wal.DefaultWatchConfig.MaxReadFrequency,
+				DrainTimeout:     wal.DefaultWatchConfig.DrainTimeout,
```

Comment:
Can we have one more config , For Ex: MaxSegmentSize which will deletes the logs when it touches the disk space size. which will be useful in some situations where the following error throws
msg=failed to write entry component=loki.write.remote err=write /tmp/agent/loki.write.remote/wal/00000569: no space left on device.

Reply:
In my opinion, the device that backs the WAL running out of space is a smell that something is happening, between:
- the agent scraping more that it can send
- the WAL volume is too small
- one should be able to make the WAL delete all segments more rapidly with `max_segment_age`, which makes sense since the WAL needs to be thought more of keeping __the last X minutes of data__ rather than __X bytes of data__

Can you share a bit more of your use case, to understand why the WAL volume is reaching it's capacity?